### PR TITLE
fix(ui-library): ensure property naming number field (831)

### PR DESCRIPTION
**Pull request by JpunktWpunkt**
related to #831 and #834

### CODE REVIEW

**Review by RubirajAccenture — APPROVED** [review]

**Review by angsherpa456 — DISMISSED** [review]

**Review by angsherpa456 — DISMISSED** [review]

**Review by angsherpa456 — DISMISSED** [review]

**Review by angsherpa456 — APPROVED** [review]

**Review by davidken91 — APPROVED** [review]

**Review by thrbnhrtmnn — CHANGES_REQUESTED** [review]
Hey @JpunktWpunkt , the changes already look really good. I would just change a few minor things, before I will approve:
1. unitPosition should per default be "prefix" and we should not offer the "none" option
2. hasUnit please update the Description in the props panel to "Choose if component has a unit.". We should be a bit more consistent with similar props like hasLabel
3. Please align the order of the props in the props panel to the Excel (I think only unitPosition and the section "Accessibility" and "Technical Attributes" need to be adjusted)

**Review by thrbnhrtmnn — DISMISSED** [review]
Storybook looks perfect now!

By the way, did you also solve https://github.com/deven-org/boiler/issues/834 with the implementation? If so, please let me know so I can close the other issue :-)

**Comment by JpunktWpunkt** [comment]
> Storybook looks perfect now!
> 
> By the way, did you also solve #834 with the implementation? If so, please let me know so I can close the other issue :-)

Yes, I did :) #834 with the last commit when I check the order I saw that name was missing :)

File: packages/ui-library/src/components/input-field-number/index.ts [hunk 1]
```diff
@@ -335,7 +337,9 @@ export class BlrInputFieldNumber extends LitElement {
                 @select=${this.handleSelect}
                 placeholder=${this.placeholder || ''}
               />
-              ${hasUnit ? html` <div class="${unitClasses}">${this.unit}</div> ` : nothing}
+              ${this.unitPosition === 'prefix' || this.unitPosition === 'suffix'
+                ? html` <div class="${unitClasses}">${this.unit}</div> `
```
Comment (angsherpa456) [inline packages/ui-library/src/components/input-field-number/index.ts:341]: Please remove the extra space on both side of html

File: packages/ui-library/src/components/input-field-number/index.ts [hunk 2]
```diff
@@ -257,40 +259,40 @@ export class BlrInputFieldNumber extends LitElement {
         [`${this.sizeVariant}`]: this.sizeVariant,
         [this.stepperVariant || 'split']: this.stepperVariant || 'split',
         'error-input': this.hasError || false,
-        'prepend': hasUnit && !!this.prependUnit,
+        // 'prepend': this.unitPosition === 'prefix' || this.unitPosition === 'suffix',
```
Comment (angsherpa456) [inline packages/ui-library/src/components/input-field-number/index.ts:262]: Can this be removed?

File: packages/ui-library/src/components/input-field-number/index.stories.ts [hunk 3]
```diff
@@ -132,7 +149,9 @@ export default {
       table: {
         category: 'Content / Settings',
       },
+      if: { arg: 'hasUnit', eq: true },
     },
+
```
Comment (angsherpa456) [inline packages/ui-library/src/components/input-field-number/index.stories.ts:154]: Please remove this additional line.

**Review by angsherpa456 — APPROVED** [review]
LGTM

**Review by thrbnhrtmnn — DISMISSED** [review]

**Review by faselbaum — APPROVED** [review]

**Review by angsherpa456 — APPROVED** [review]
LGTM

**Pull request merged**